core/scripts/gateway/workflow/upload_workflow.go

```diff
@@ -0,0 +1,106 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/joho/godotenv"
+
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/api"
+)
+
+func main() {
+	gatewayURL := flag.String("gateway_url", "http://localhost:5002", "Gateway URL")
+	privateKey := flag.String("private_key", "65456ffb8af4a2b93959256a8e04f6f2fe0943579fb3c9c3350593aabb89023f", "Private key to sign the message with")
```

Comment:
Am I understanding this correctly? this is the workflow owner's private key and some addresses/private keys are allowlisted in the Gateway DON.

Reply:
Yes - owner's private key used to sign messages to Gateway.
Yes - there needs to be some kind of allowlist or an account system on Gateway's side (of addresses, not private keys).